File: 0700_Search in a Binary Search Tree/SearchinaBinarySearchTree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode SearchBST(TreeNode root, int val) {
        if(root == null) return null;
        if(root.val == val) return root;
        if(root.val > val){
            return SearchBST(root.left,val);
        }else{
            return SearchBST(root.right,val);
        }
    }
}

File: 0700_Search in a Binary Search Tree/SearchinaBinarySearchTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode searchBST(TreeNode root, int val) {
        if(root == null) return null;
        TreeNode p = root;
        while(p!=null){
            if(p.val == val) return p;
            if(p.val > val){
                p = p.left;
            }else{
                p = p.right;
            }
        }
        
        return null;
    }
}